### ISSUE TITLE
One broken bike rack makes racked vehicle broken

### ISSUE THREAD

**Issue by dev-dsp** [CleverRaven/Cataclysm-DDA#76613]
### Describe the bug

Can't unrack a vehicle after one of bike racks was broken (by Zs)

### Attach save file

[Nairn-trimmed.tar.gz](https://github.com/user-attachments/files/17090423/Nairn-trimmed.tar.gz)

Here, I've installed missing bike racks, but it didn't help. 'e' on bike rack tile says nothing's racked, but attempting to fix folding wire basket on my 'booter' says it should be unracked before starting works.the the 

### Steps to reproduce

1. Have vehicle racked
2. Break rack
3. Racked vehicle is stuck

### Expected behavior

Working rack / ability to fix

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Gentoo; Description: Gentoo Linux; Release: 2.15; Codename: n/a; 
- Game Version: cdda-experimental-2024-09-19-0617 e00a779 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex]
]

**Comment by PatrikLundell** [comment]
Sounds like logic that can break, yes. I'd try to repair the rack and then save and reload the game, in the hope that maybe loading restores some of the missing data. It's a low probability, but still worth a try.

**Comment by PatrikLundell** [comment]
I don't think this can be fixed via the debug menu. In theory it should be possible to hack the save files, but that would require finding the correct file, finding the vehicle in it, and figuring out how the data should be changed, and then change the file without making any mistakes.

It should be possible to debug create a new vehicle like the racked one if it is a standard type and then manually smash the racked parts of the current racked vehicle, but that still wouldn't guarantee the rack won't be bugged and still think something is attached, although removing the racks and reinstalling them might work.
I don't know if the debug menu can be used to destroy vehicle parts (so smashing wouldn't be needed).
Anyway, I'd perform any such experiments on a copy of the save so failed attempts aren't accidentally overwriting it (in particular if automatic saves are enabled).

**Comment by TheHopelessThrumbo** [comment]
I ended up randomly encountering this issue in stable Herbert release, and decided to test it on the latest Experimental build (e201310)... the issue is still present, and reproducable under the exact conditions specified.

[Neville-trimmed.tar.gz](https://github.com/user-attachments/files/18736398/Neville-trimmed.tar.gz)

![Image](https://github.com/user-attachments/assets/1518a511-dc88-4113-95be-a43dc6473d3e)
![Image](https://github.com/user-attachments/assets/1fbc65d0-cf2f-4159-a9d1-a9b2ce8e634b)
